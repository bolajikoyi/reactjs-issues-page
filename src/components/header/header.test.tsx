import React from 'react';
import { render, screen, cleanup } from '@testing-library/react';
import ReactDOM from 'react-dom';
import '@testing-library/jest-dom/extend-expect';
import renderer from 'react-test-renderer';
import Header from './header';

afterEach(cleanup);

describe('Header Component', ()=>{
    test('Expect componet to be created', () => {
        expect(Header).toBeTruthy()
    });
    
    test('renders without crashing', () => {
        const div = document.createElement('div');
        ReactDOM.render(<Header />, div);
    });

    test("Should Contain Text 'Search or jump to...'", async () => {
        render(<Header />);
        screen.getByText('Search or jump to...');
    });

    test("Should Contain span tag for 'Search or jump to...' text", async () => {
        render(<Header />);
        expect(screen.getByText("Search or jump to...").tagName).toBe("SPAN");
    });

    test("Should Contain ClassName", async () => {
        render(<Header />);
        expect(screen.getByText("/").className).toBe(
          "Header-search-button"
        );  
      });
      
    test('matches snapshot', ()=>{
        const tree = renderer.create(<Header/>).toJSON();
        expect(tree).toMatchSnapshot();
    })
})

