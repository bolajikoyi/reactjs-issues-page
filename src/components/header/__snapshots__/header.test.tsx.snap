// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header Component matches snapshot 1`] = `
<div>
  <div
    className="Header"
  >
    <div
      className="Header-left"
    >
      <i
        className="header fab fa-github fa-2x"
      />
      <div
        className="Header-search"
      >
        <span>
          Search or jump to...
        </span>
        <div
          className="Header-search-button"
        >
          /
        </div>
      </div>
      <div>
        <span>
          Pull requests
        </span>
        <span>
          Issues
        </span>
        <span>
          Marketplace
        </span>
        <span>
          Explore
        </span>
      </div>
    </div>
    <div
      className="Header-right"
    >
      <i
        className="header far fa-bell"
      />
      <span>
        + 
        <i
          className="header fas fa-caret-down"
        />
      </span>
      <span>
        <i
          className="header fas fa-user"
        />
         
        <i
          className="header fas fa-caret-down"
        />
      </span>
    </div>
  </div>
</div>
`;
